app/grouper_syncer.pl

```diff
@@ -0,0 +1,327 @@
+	my $res = $ua->request($r);
+	
+	log_debug("Results " . Dumper($res));
+	my $obj = decode_json($res->content)->{'WsGetSubjectsResults'};
+
+	log_debug("Attribute Values: " . Dumper($obj->{'wsSubjects'}->[0]->{'attributeValues'}));
+	log_debug("Attribute Names: " . Dumper($obj->{'subjectAttributeNames'}));
+
+	my $attrs = {};
+	next if !defined($obj->{'wsSubjects'}->[0]->{'attributeValues'});
+	for(my $i=0;$i<=scalar(@{$obj->{'wsSubjects'}->[0]->{'attributeValues'}});$i++){
+	    $attrs->{ $obj->{'subjectAttributeNames'}->[$i] } = $obj->{'wsSubjects'}->[0]->{'attributeValues'}->[$i];
+	}
+	
+	log_debug("User details: " .  Dumper($attrs));
+
+	push(@grouper_users,{ id => $gu->{'id'},
+			      first_name => $attrs->{'givenname'},
+			      last_name => $attrs->{'sn'},
+			      username => $attrs->{'edupersonprincipalname'},
+			      email => $attrs->{'mail'} } );
+	
+    }
+    
+    return \@grouper_users;
+}
+
+
+
+sub fetch_config{
+    my $config_file = shift;
+    log_debug("Config File: " . $config_file); 
+    my $config = GRNOC::Config->new( config_file => $config_file);
+    
+    my $res = {};
+    
+    my $grouper = {};
+    my $oess = {};
+
+    $grouper->{'user'} = $config->get('/config/grouper/@user')->[0];
+    $grouper->{'url'} = $config->get('/config/grouper/@url')->[0];
+    $grouper->{'password'} = $config->get('/config/grouper/@password')->[0];
+
+    $oess->{'user'} = $config->get('/config/oess/@user')->[0];
+    $oess->{'url'} = $config->get('/config/oess/@url')->[0];
+    $oess->{'password'} = $config->get('/config/oess/@password')->[0];
+
+    $res->{'grouper'} = $grouper;
+    $res->{'oess'} = $oess;
+
+    return $res;
+}
+
+sub connect_to_oess{
+    my $config = shift;
+
+    my $wsc = GRNOC::WebService::Client->new( url => $config->{'oess'}->{'url'} . "/admin/admin.cgi",
+					      uid => $config->{'oess'}->{'user'},
+					      passwd => $config->{'oess'}->{'password'},
+					      realm => $config->{'oess'}->{'realm'},
+					      verify_hostname => 0,
+					      debug => 0
+	);
+
+    return $wsc;
+
+    
+}
+
+sub make_group_request{    
+
+    my %params = @_;
+
+    my $url = $params{'url'};
+    my $user = $params{'user'};
+    my $password = $params{'password'};
+    my $stem = $params{'stem'};
+
+    my $ua = LWP::UserAgent->new;
+
+    my $obj = {'WsRestFindGroupsRequest' => {'actAsSubjectLookup' => {'subjectId' => 'oess_provision'},
+	       'wsQueryFilter' => {'queryFilterType' => 'FIND_BY_STEM_NAME', 'stemName' => $stem}}};
+    my $str = encode_json($obj);
+    #log_debug("JSON STring: $str");
+    #my $str = '{
+#  "WsRestFindGroupsRequest":{
+#    "actAsSubjectLookup":{
+#      "subjectId":"oess_provision"
+#    },
+#    "wsQueryFilter":{
+#      "queryFilterType":"FIND_BY_STEM_NAME",
+#      "stemName":"app:oess:service:workgroups"
+#    }
+#  }
+#}';
+
+    my $h = HTTP::Headers->new(
+	Content_Length => length($str),
+	Content_Type => 'text/x-json' );
```

Comment:
Correct content type?

Reply:
Yes actually this is!  Its required by the Grouper API